tests/bson/bson-fromPHP-006.phpt

```
@@ -0,0 +1,50 @@
+--TEST--
+BSON\fromPHP(): PHP documents with null bytes in field name
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+echo "\nTesting array with one leading null byte in field name\n";
+hex_dump(fromPHP(["\0" => 1]));
```

I agree that should throw an exception. Silent data truncation is scary.